File: sdk/nodejs/src/resources/cloud_gpu.js
"use strict";

const { BaseResource } = require("./base");

class CloudGpuResource extends BaseResource {
  async reboot(serviceId, data) {
    return this._http.put("/api/service/{id}/vms/reboot", data);
  }

  async stop(serviceId, data) {
    return this._http.put("/api/service/{id}/vms/stop", data);
  }

  async start(serviceId, data) {
    return this._http.put("/api/service/{id}/vms/start", data);
  }

  async getFirewall(serviceId) {
    return this._http.get("/api/service/{id}/vms/firewall");
  }

  async createFirewallRule(serviceId, data) {
    return this._http.post("/api/service/{id}/vms/firewall", data);
  }

  async deleteFirewallRule(serviceId, position) {
    return this._http.delete(`/api/service/{id}/vms/firewall/${position}`);
  }

}

module.exports = { CloudGpuResource };

File: sdk/csharp/Resources/CloudGpu.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace HiTechCloud.SDK.Resources;

public class CloudGpu : BaseResource
{
    public CloudGpu(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> RebootAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PutAsync(string.Format("/api/service/{id}/vms/reboot", id), data);
    }

    public async Task<Dictionary<string, object?>> StopAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PutAsync(string.Format("/api/service/{id}/vms/stop", id), data);
    }

    public async Task<Dictionary<string, object?>> StartAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PutAsync(string.Format("/api/service/{id}/vms/start", id), data);
    }

    public async Task<Dictionary<string, object?>> FirewallAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/vms/firewall", id));
    }

    public async Task<Dictionary<string, object?>> CreateFirewallAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/vms/firewall", id), data);
    }

    public async Task<Dictionary<string, object?>> DeleteFirewallAsync(int id, int position)
    {
        return await Http.DeleteAsync(string.Format("/api/service/{id}/vms/firewall/{position}", id, position));
    }
}
